### DISCUSSION TITLE
Batterie Status (Kalibrierung)

### DISCUSSION THREAD

**Discussion by OnkelFD** [aruttkamp/ioBroker.rct#69] [Fragen & Antworten] [answered]
Hallo,

ich habe mir das Objekt "battery.bat_status" ein wenig genauer angeschaut und folgendes herausgefunden.
Folgenden Werte konnte ich in Verbindung mit der Anzeige in der App ermitteln.

Objekt -> App

0 -> charge/discharge
8 -> charge (calib)
1024 -> discharge (calib)
2028 -> balancing

Ich hoffe ich kann damit einigen helfen bzw. Wenn dies offiziell in den Adapter einfließen könnte wäre das bestimmt auch hilfreich.
Hat von euch auch schon jemand diese Zustände ermitteln können oder noch weitere?

Gruß

**Comment by aruttkamp** [comment]
Vielen Dank ! Das klingt tatsächlich interessant ! 
Aktuell gibt der Adapter ja die Werte (0/8/1024/2024) aus. 

Reicht es deiner Meinung nach aus die States zu dokumentieren oder würdest Du dir wünschen 
dass der Adapter die Klartexte der States ausgibt ? 

Ich würde aktuell zur Dokumentation tendieren.

Andreas

**Comment by OnkelFD** [comment]
Also ich würde sagen wenn es in der Doku aufgelistet ist würde das reichen die weitere Verarbeitung kann man dann selbst über den Broker machen.

**Comment by aruttkamp** [comment]
OnkelFD - Kann es sein , dass es 
2048 -> balancing
heissen muss ?

**Comment by OnkelFD** [comment]
🫣 Ja da hat sich wohl ein kleiner Fehler eingeschlichen 2048 ist die richtige Zahl...

**Comment by OnkelFD** [comment]
Habe mir diesbezüglich ein Script gebastelt welches das Kalibrier Datum anzeigt und das Laden/Entladen bzw. Kalibrieren/Balancing,
wenn jemand Interesse daran hat.

Daten werden unter "javascript.0.Eigene_Datenpunkte.RCT" abgelegt kann aber auch geändert/angepasst werden.

notwendige Datenpunkte sind im RCT Adapter einzutragen:
bat_status
battery_power
bat_next_calib_date


[RCT Calib Datum.txt](https://github.com/aruttkamp/ioBroker.rct/files/13640695/RCT.Calib.Datum.txt)

**Comment by NCIceWolf** [comment, marked as answer]
Hab das Thema mal so in die Doku eingefügt, ich denke mit den States kann man sogar mehr anfangen als wenn man das als Text zurück bekommt. @OnkelFD hat dazu ja sogar ein Skript beigesteuert 👍

**Comment by aruttkamp** [comment]
Danke an NCIceWolf für das ergänzen der Doku und OnkelFD für das Skript !

**Comment by OnkelFD** [comment]
Also ich habe jetzt noch zwei weitere Zustände gefunden 3 & 5.

Kann sie aber nicht komplett/richtig zuordnen.

Die Zahl 3 und 5 das müsste beim Update entstanden sein, erst ist die 3 aufgezeichnet worden und kurz danach ( ca.1min ) die 5.

Könnte mir folgendes vorstellen:
3 -> Update ?
5 -> Start ?

Wenn jemand von euch das Update noch nicht gemacht hat und dies überprüfen könnte wäre Super.

**Reply by NCIceWolf** [reply]
Sorry, bin leider schon durch und hab die Zustände nicht in der Influx protokolliert.
Das könnte hinhauen, aber sind die für den "täglichen" Betrieb interessant?

**Reply by OnkelFD** [reply]
Ist meiner Meinung nach auch nicht für den täglichen Betrieb von Belang das stimmt schon, allerdings ist es dennoch gut zu wissen.
Je mehr man weis desto besser 😉 (Wissen schadet nicht).

**Comment by NCIceWolf** [comment]
Ich denke das Thema ist soweit abgeschlossen, das Skript tut sauber seinen Dienst.
